nixos/modules/services/networking/iwd.nix

```diff
@@ -74,7 +175,27 @@ in
       wantedBy = [ "multi-user.target" ];
       restartTriggers = [ configFile ];
     };
+
+    systemd.services.iwd.preStart = mkIf (cfg.networks != {}) (let
+      ssids = lib.flip lib.mapAttrsToList predefinedNetworks (file: content: {
+        inherit (content) file;
+        local = encodeIfNeeded file content.extension;
+      });
+    in ''
+      # Remove all network-configs from `/var/lib/iwd` that are symlinks to a store-path,
```

Comment:
It looks like *iwd* will most likely replace these network symlinks during runtime, as *iwd* expects to be able to update these files at runtime.

For WPA2 if `passphrase` is defined while `PreSharedKey` is not, *iwd* recreates the file, adding the `PreSharedKey` field in the security section.

WPA3 is similar, with it adding `SAE-PT-Group19` and `SAE-PT-Group20` fields.


Maybe the directory should be wiped on service restart?